opm/parser/eclipse/EclipseState/Schedule/Schedule.hpp

```diff
@@ -418,6 +445,56 @@ namespace Opm
                            const UnitSystem& unit_system,
                            std::vector<std::pair<const DeckKeyword*, size_t > >& rftProperties);
         void addWellGroupEvent(const std::string& wellGroup, ScheduleEvents::Events event, size_t reportStep);
+
+        template<template<class, class> class Map, class Type, class Key>
+        std::pair<std::vector<Type>, std::vector<std::pair<Key, std::vector<int>>>>
+        splitDynMap(const Map<Key, Opm::DynamicState<Type>>& map)
+        {
+            // we have to pack the unique ptrs separately, and use an index map
+            // to allow reconstructing the appropriate structures.
+            std::vector<std::pair<Key, std::vector<int>>> asMap;
+            std::vector<Type> unique;
+            for (const auto& it : map) {
+                for (const auto& w : it.second.data()) {
+                    if (std::find(unique.begin(), unique.end(), w) == unique.end())
+                        unique.push_back(w);
+                }
+            }
+            for (const auto& it : map) {
```

Comment:
Somehow the second iteration over the map seems superfluous (and might be expensive due to the find). Is there a special reason not to do
```
std::vector<int> idxVec;

            for (const auto& it : map) {
                for (const auto& w : it.second.data()) {
                    auto candidate = std::find(unique.begin(), unique.end(), w);
                    auto idx = candidate - unique.begin();
                    if (candidate == unique.end())
                   {
                        unique.push_back(w);
                        idx = unique.size()-1;
                    }
                   idxVec.push_back(idx);
                }
                idxVec.push_back(it.second.initialRange());
                asMap.push_back(std::make_pair(it.first, idxVec));
            }

Reply:
no reason, nice catch, done.